### PR TITLE
msg-nommu: Fix rid being allocaded in the wrong place

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-606

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: proc/msg-nommu.c [hunk 1]
```diff
@@ -107,19 +107,23 @@ int proc_recv(u32 port, msg_t *msg, msg_rid_t *rid)
...
 
 	if (p->closed) {
 		/* Port is being removed */
 		if (kmsg != NULL) {
 			kmsg->state = msg_rejected;
-			LIST_REMOVE(&p->kmessages, kmsg);
 			proc_threadWakeup(&kmsg->threads);
 		}
 
 		hal_spinlockClear(&p->spinlock, &sc);
```
Comment (lukileczo) [inline proc/msg-nommu.c:122]: I think spinlock clear may be safely moved earlier - `kmsg` is no longer accessible through `p`:
```suggestion
	if (p->closed) {
		/* Port is being removed */
		hal_spinlockClear(&p->spinlock, &sc);
		if (kmsg != NULL) {
			kmsg->state = msg_rejected;
			proc_threadWakeup(&kmsg->threads);
		}
```

Comment (agkaminski) [reply proc/msg-nommu.c:122]: I kinda agree, but sender still can access `kmsg` if it spuriously wakes up, same reason in lines 129 and 132

**Comment by lukileczo** [comment]
Shouldn't there be an early return path if we get `EINTR` from `proc_threadWaitInterruptible` but the message has not arrived (i.e. is `NULL`)? Now we can get to `proc_portRidAlloc` with `NULL`

**Comment by agkaminski** [comment]
> Shouldn't there be an early return path if we get `EINTR` from `proc_threadWaitInterruptible` but the message has not arrived (i.e. is `NULL`) and port is not closed? Now we can get to `proc_portRidAlloc` with `NULL` Should've looked at the first review 💀

You're right, should be fixed now

**Review by lukileczo — APPROVED** [review]
LGTM

**Pull request merged**